fix: 663 create idea popup

TheMeerCat opened the pull request:
#663

piperchester commented on frontend/src/components/dashboard/Dashboard.js:
```diff
@@ -14,13 +14,16 @@ import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
 import UserImage from "./UserImage";
+import CreateIdeaDialog from "../ideas/createIdea/CreateIdeaDialog";
+import { getUserOrganizations } from "../../../public/lib/organizationOperations";
+import Cookies from "universal-cookie";
```
Nit: do you mind colocating this 3rd party import to the others, up with line 18? And then delineating relative imports and 3rd party with a newline? I find this helps aid with readability some :-)

marketajancova replied:
Sure, no problem. :-)

piperchester commented on frontend/src/components/hub/HubContent.js:
```diff
@@ -133,7 +135,13 @@ export default function HubContent({
                 {user ? (
-                  <Dashboard location={location} headline={headline} hubUrl={hubUrl} />
+                  <Dashboard
+                    allHubs={allHubs}
+                    hubData={hubData}
+                    location={location}
+                    headline={headline}
+                    hubUrl={hubUrl}
```
Nit: do you mind alphabetizing these? If you're using VS Code, this is really easy by pressing `cmd+shift+p` and then typing "sort".

marketajancova replied:
Sure, no problem. Personally, I don't alphabetize shortcuts but if it is our standard, I will definitely follow it. :-)

marketajancova replied:
Hmmm I've checked it in code and we have "hubUrl" everywhere. Do we really want to do alphabetize it?

positiveimpact replied:
I feel like doing this manually doesn't make too much sense at this point since we've already not been doing it in the rest of the codebase. Instead we should probably update our configuration for `prettier` so this gets done automatically.

piperchester commented on frontend/src/components/dashboard/Dashboard.js:
```diff
@@ -161,16 +164,24 @@ const DropDownList = ({ buttonRef, handleOpen, handleClose, items, open }) => {
+            <Link
+              key={item.url_slug}
+              href={item.url_slug}
+              onClick={() => handleClick(item.onClick)}
```
Is there a benefit here to using inline arrow functions over the bound handler?

marketajancova replied:
I am not sure whether I understand what you mean but `onClick` requires `Function` type and doing `onClick={() => handleClick(item.onClick)}` would just call `handleClick(item.onClick)` during rendering and insert the output as the parameter. I could insert `onClick={handleClick}` without the arrow but then I would not be able to pass the `item.onClick` parameter. Do you mean some other way? :-)

piperchester replied:
Nope that makes sense! Thanks for explaining :-)

marketajancova commented:
Thank you for your review/comments @piperchester :-)

positiveimpact commented on frontend/src/components/dashboard/Dashboard.js:
```diff
@@ -182,9 +193,21 @@ const DropDownList = ({ buttonRef, handleOpen, handleClose, items, open }) => {
+  const [isCreateIdeaOpen, setCreateIdeaOpen] = useState(false);
+  const token = new Cookies().get("token");
+
+  useEffect(async function () {
+    if (userOrganizations === null) {
+      setUserOrganizations("");
+      const userOrgsFromServer = await getUserOrganizations(token, locale);
+      setUserOrganizations(userOrgsFromServer || []);
+    }
+  });
```
Does this have to run multiple times? Because if it only has to run once, I'd add the following suggestion:
```suggestion
  useEffect(async function () {
    if (userOrganizations === null) {
      setUserOrganizations("");
      const userOrgsFromServer = await getUserOrganizations(token, locale);
      setUserOrganizations(userOrgsFromServer || []);
    }
  }, []);
```

marketajancova replied:
True, I fixed it

positiveimpact commented on frontend/src/components/dashboard/Dashboard.js:
```diff
@@ -161,16 +164,24 @@ const DropDownList = ({ buttonRef, handleOpen, handleClose, items, open }) => {
-            <Link key={item.url_slug} href={item.url_slug} onClick={handleClick}>
+            <Link
+              key={item.url_slug}
+              href={item.url_slug}
```
The href here needs to be removed if item.url_slug is empty. Otherwise, it will open the idea popup for a second and then just refresh the current page

marketajancova replied:
For the idea popup, the href is empty string. I tested it and it does not redirect you anywhere

positiveimpact reviewed (commented):
LGTM except my comments. One thing that would be ideal would be if it also switched tab to the ideas tab automatically.

piperchester reviewed (approved):
Nice work!

TheMeerCat commented:
> LGTM except my comments. One thing that would be ideal would be if it also switched tab to the ideas tab automatically.

I changed it based on your suggestion. Unfortunately, the redirect at the end is a bit problematic as we have to load "ideas" data. Do you think we are ok with refreshing the page at the end?

positiveimpact commented:
> > LGTM except my comments. One thing that would be ideal would be if it also switched tab to the ideas tab automatically.
> 
> I changed it based on your suggestion. Unfortunately, the redirect at the end is a bit problematic as we have to load "ideas" data. Do you think we are ok with refreshing the page at the end?

I'd merge it without the tab switch for now. It would be possible to switch to the tab and load the data in the background without refreshing. But since tab switching is handled in BrowseContent that would be a lot of refactoring to be able to trigger it without a redirect from the dashboard component. And after creating an idea you're redirected to the ideas tab anyways so from my side this can be merged as is.

marketajancova commented:
> > > LGTM except my comments. One thing that would be ideal would be if it also switched tab to the ideas tab automatically.
> > 
> > 
> > I changed it based on your suggestion. Unfortunately, the redirect at the end is a bit problematic as we have to load "ideas" data. Do you think we are ok with refreshing the page at the end?
> 
> I'd merge it without the tab switch for now. It would be possible to switch to the tab and load the data in the background without refreshing. But since tab switching is handled in BrowseContent that would be a lot of refactoring to be able to trigger it without a redirect from the dashboard component. And after creating an idea you're redirected to the ideas tab anyways so from my side this can be merged as is.

Yeh, I do agree. So I reverted it. All good now?

positiveimpact commented:
> > > > LGTM except my comments. One thing that would be ideal would be if it also switched tab to the ideas tab automatically.
> > > 
> > > 
> > > I changed it based on your suggestion. Unfortunately, the redirect at the end is a bit problematic as we have to load "ideas" data. Do you think we are ok with refreshing the page at the end?
> > 
> > 
> > I'd merge it without the tab switch for now. It would be possible to switch to the tab and load the data in the background without refreshing. But since tab switching is handled in BrowseContent that would be a lot of refactoring to be able to trigger it without a redirect from the dashboard component. And after creating an idea you're redirected to the ideas tab anyways so from my side this can be merged as is.
> 
> Yeh, I do agree. So I reverted it. All good now?

:+1:
